specification/sections/01-flag-evaluation.md

```diff
@@ -353,9 +353,15 @@ Configuration code includes code to set the provider, instantiate providers, and
 
 #### Requirement 1.4.11
 
-> In the case of abnormal execution, the client **SHOULD** log an informative error message.
+> Methods, functions, or operations on the client **SHOULD NOT** write log messages.
```

Comment:
Couldn't we go for MAY NOT here?
I think this is fine to enforce.

Reply:
My reasoning is that there's some amount of "gray area" here. For example. I left some logging for unhandled errors in event handlers and hooks, which are technically part of the lifecycle, but seem to me to be a lot different than something like a FLAG_NOT_FOUND in an management system (more like a coding error). RFC2113 feels right to me for this, because it allows deviation, but only with thoughtful consideration:

https://datatracker.ietf.org/doc/html/rfc2119#section-4

> This phrase, or the phrase "NOT RECOMMENDED" mean that
   there may exist valid reasons in particular circumstances when the
   particular behavior is acceptable or even useful, but the full
   implications should be understood and the case carefully weighed
   before implementing any behavior described with this label.
   
I think for logging, this is good. You _can_ log in the hot path, but it's discouraged and should be done very carefully to avoid logspam.

I think if we do MUST NOT, our list of exceptions will be growing forever for valid reasons.

Reply:
Okay makes sense to me!